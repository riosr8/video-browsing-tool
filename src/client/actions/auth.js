import axios from 'axios';
import * as actionTypes from './actionTypes';

export const loginStart = () => {
  return {
    type: actionTypes.LOGIN_START,
  };
};

export const loginSuccess = (token, userId) => {
  return {
    type: actionTypes.LOGIN_SUCCESS,
    idToken: token,
    userId,
  };
};

export const loginFail = error => {
  return {
    type: actionTypes.LOGIN_FAIL,
    error,
  };
};

export const registrationStart = () => {
  return {
    type: actionTypes.REGISTRATION_START,
  };
};

export const registrationSuccess = () => {
  return {
    type: actionTypes.REGISTRATION_SUCCESS,
  };
};

export const registrationFail = error => {
  return {
    type: actionTypes.REGISTRATION_FAIL,
    error,
  };
};

export const logout = () => {
  localStorage.removeItem('token');
  localStorage.removeItem('expirationDate');
  localStorage.removeItem('userId');
  return {
    type: actionTypes.AUTH_LOGOUT,
  };
};

export const beginFormErrorChecking = () => {
  return {
    type: actionTypes.BEGIN_FORM_ERROR_CHECKING,
  };
};

export const setFormErrorStart = () => {
  return {
    type: actionTypes.SET_FORM_ERROR_START,
  };
};

export const setFormErrorEnd = error => {
  return {
    type: actionTypes.SET_FORM_ERROR_END,
    error,
  };
};

export const resetRegistrationCompleted = () => {
  return {
    type: actionTypes.RESET_REGISTRATION_COMPLETED,
  };
};

export const setError = error => {
  return dispatch => {
    dispatch(setFormErrorStart());
    dispatch(setFormErrorEnd(error));
  };
};

export const auth = (email, password) => {
  return async dispatch => {
    dispatch(loginStart());
    const authData = {
      email,
      password,
    };
    await axios
      .post('/auth/login', authData)
      .then(response => {
        const expirationDate = new Date(new Date().getTime() + response.data.expiresIn);
        localStorage.setItem('token', response.data.token);
        localStorage.setItem('expirationDate', expirationDate);
        localStorage.setItem('userId', response.data.email);
        dispatch(loginSuccess(response.data.token, response.data.email));
      })
      .catch(err => {
        dispatch(loginFail(err.response.data.message));
      });
  };
};

export const register = (email, password) => {
  return async dispatch => {
    dispatch(registrationStart());
    const registrationData = {
      email,
      password,
    };
    await axios
      .post('/register/newUser', registrationData)
      // eslint-disable-next-line no-unused-vars
      .then(response => {
        dispatch(registrationSuccess());
      })
      .catch(err => {
        dispatch(registrationFail(err.response.message));
      });
  };
};

export const setAuthRedirectPath = path => {
  return {
    type: actionTypes.SET_AUTH_REDIRECT_PATH,
    path,
  };
};

export const checkAuthTimeout = expirationTime => {
  return dispatch => {
    setTimeout(() => {
      dispatch(logout());
    }, expirationTime * 1000);
  };
};

export const checkAuthState = () => {
  return dispatch => {
    const token = localStorage.getItem('token');
    if (!token) {
      dispatch(logout());
    } else {
      const expirationDate = new Date(localStorage.getItem('expirationDate'));
      if (expirationDate <= new Date()) {
        dispatch(logout());
      } else {
        const userId = localStorage.getItem('userId');
        dispatch(loginSuccess(token, userId));
        dispatch(checkAuthTimeout((expirationDate.getTime() - new Date().getTime()) / 1000));
      }
    }
  };
};
