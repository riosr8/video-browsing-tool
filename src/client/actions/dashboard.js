import axios from 'axios';
import * as actionTypes from './actionTypes';

// FETCH VIDEO ACTION CREATORS

export const fetchVideosStart = () => {
  return {
    type: actionTypes.FETCH_VIDEOS_START,
  };
};

export const fetchVideosSuccess = (userVideos, totalPages) => {
  return {
    type: actionTypes.FETCH_VIDEOS_SUCCESS,
    userVideos,
    totalPages,
  };
};

export const fetchVideosFail = error => {
  return {
    type: actionTypes.FETCH_VIDEOS_FAIL,
    error,
  };
};

export const uploadVideoStart = () => {
  return {
    type: actionTypes.UPLOAD_VIDEO_START,
  };
};

export const uploadVideoSuccess = success => {
  return {
    type: actionTypes.UPLOAD_VIDEO_SUCCESS,
    success,
  };
};

export const uploadVideoFail = error => {
  return {
    type: actionTypes.UPLOAD_VIDEO_FAIL,
    error,
  };
};

export const requestProcessingStart = () => {
  return {
    type: actionTypes.REQUEST_PROCESSING_START,
  };
};

export const requestProcessingSuccess = message => {
  return {
    type: actionTypes.REQUEST_PROCESSING_SUCCESS,
    success: message,
  };
};

export const requestProcessingFail = error => {
  return {
    type: actionTypes.REQUEST_PROCESSING_FAIL,
    error,
  };
};

export const fetchAlgorithmsStart = () => {
  return {
    type: actionTypes.FETCH_ALGORITHMS_START,
  };
};

export const fetchAlgorithmsSuccess = algorithms => {
  return {
    type: actionTypes.FETCH_ALGORITHMS_SUCCESS,
    algorithms,
  };
};

export const fetchAlgorithmsFail = () => {
  return {
    type: actionTypes.FETCH_ALGORITHMS_FAIL,
  };
};

export const deleteVideoStart = () => {
  return {
    type: actionTypes.DELETE_VIDEO_START,
  };
};

export const deleteVideoSuccess = success => {
  return {
    type: actionTypes.DELETE_VIDEO_SUCCESS,
    success,
  };
};

export const deleteVideoFail = error => {
  return {
    type: actionTypes.DELETE_VIDEO_FAIL,
    error,
  };
};

export const fetchResultsStart = () => {
  return {
    type: actionTypes.FETCH_RESULTS_START,
  };
};

export const fetchResultsSuccess = results => {
  return {
    type: actionTypes.FETCH_RESULTS_SUCCESS,
    results,
  };
};

export const fetchResultsFail = error => {
  return {
    type: actionTypes.FETCH_RESULTS_FAIL,
    error,
  };
};

export const clearMessage = () => {
  return {
    type: actionTypes.CLEAR_MESSAGE,
  };
};

export const setActivePage = activePage => {
  return {
    type: actionTypes.SET_ACTIVE_PAGE,
    activePage,
  };
};

export const setSortKey = sortKey => {
  return {
    type: actionTypes.SET_SORT_KEY,
    sortKey,
  };
};

export const setSortOrder = sortOrder => {
  return {
    type: actionTypes.SET_SORT_ORDER,
    sortOrder,
  };
};

export const getResults = (token, uuid) => {
  return async dispatch => {
    dispatch(fetchResultsStart());
    await axios
      .post(
        '/api/results',
        {
          videoUuid: uuid,
        },
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        },
      )
      .then(res => {
        dispatch(fetchResultsSuccess(res.data));
      })
      .catch(err => {
        dispatch(fetchResultsFail(err.response.data.message));
      });
  };
};

export const getVideos = (token, pageNo, sortKey, sortOrder) => {
  return async dispatch => {
    dispatch(fetchVideosStart());
    await axios
      .get(`/api/listPagination?pageNo=${pageNo}&sortVal=${sortKey}&order=${sortOrder}`, {
        headers: {
          'Access-Control-Allow-Origin': '*',
          'Access-Control-Allow-Methods': 'GET,PUT,POST,DELETE,PATCH,OPTIONS',
          Authorization: `Bearer ${token}`,
        },
      })
      .then(res => {
        dispatch(fetchVideosSuccess(res.data.results, res.data.totalCount));
      })
      .catch(err => {
        dispatch(fetchVideosFail(err.response.data.message));
      });
  };
};

export const getAlgorithms = () => {
  return async dispatch => {
    dispatch(fetchAlgorithmsStart());
    await axios
      .get('/noAuth/algorithmsAvailable')
      .then(res => {
        dispatch(fetchAlgorithmsSuccess(res.data.algorithms));
      })
      .catch(err => {
        dispatch(fetchAlgorithmsFail(err.response.data.message));
      });
  };
};

export const uploadVideo = (data, token, activePage, sortKey, sortOrder) => {
  return async dispatch => {
    dispatch(uploadVideoStart());
    await axios
      .post('/api/upload', data, {
        headers: {
          'Access-Control-Allow-Origin': '*',
          'Access-Control-Allow-Methods': 'GET,PUT,POST,DELETE,PATCH,OPTIONS',
          Authorization: `Bearer ${token}`,
        },
      })
      .then(res => {
        dispatch(uploadVideoSuccess(`${res.data.videoName} successfully uploaded.`));
      })
      .then(() => {
        dispatch(getVideos(token, activePage, sortKey, sortOrder));
      })
      .catch(err => {
        dispatch(uploadVideoFail(err.response.data.message));
      });
  };
};

export const sendProcessingRequest = (token, selectedAlgorithms, selectedVideoUUID) => {
  return async dispatch => {
    dispatch(requestProcessingStart());
    await axios
      .post(
        '/api/algorithmInput',
        {
          videoUuid: selectedVideoUUID,
          algorithmicInput: selectedAlgorithms,
        },
        {
          headers: {
            'Access-Control-Allow-Origin': '*',
            'Access-Control-Allow-Methods': 'GET,PUT,POST,DELETE,PATCH,OPTIONS',
            Authorization: `Bearer ${token}`,
          },
        },
      )
      .then(res => {
        dispatch(requestProcessingSuccess(res.data.message));
      })
      .catch(err => {
        dispatch(requestProcessingFail(err.response.data.message));
      });
  };
};

export const deleteVideo = (token, uuid, activePage, sortKey, sortOrder) => {
  return async dispatch => {
    dispatch(deleteVideoStart());
    await axios
      .post(
        '/api/delete',
        {
          videoUuid: uuid,
        },
        {
          headers: {
            'Access-Control-Allow-Origin': '*',
            'Access-Control-Allow-Methods': 'GET,PUT,POST,DELETE,PATCH,OPTIONS',
            Authorization: `Bearer ${token}`,
          },
        },
      )
      .then(res => {
        dispatch(deleteVideoSuccess(res.data.message));
      })
      .then(() => {
        dispatch(getVideos(token, activePage, sortKey, sortOrder));
      })
      .catch(err => {
        dispatch(deleteVideoFail(err.response.data.message));
      });
  };
};
