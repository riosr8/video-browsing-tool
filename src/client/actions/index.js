export {
  auth,
  setAuthRedirectPath,
  checkAuthState,
  logout,
  register,
  setError,
  beginFormErrorChecking,
  resetRegistrationCompleted,
} from './auth';

export {
  getVideos,
  uploadVideo,
  sendProcessingRequest,
  deleteVideo,
  getAlgorithms,
  getResults,
  clearMessage,
  setActivePage,
  setSortKey,
  setSortOrder,
} from './dashboard';

export { pollStatus } from './processStatus';
