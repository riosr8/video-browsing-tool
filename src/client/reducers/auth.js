import * as actionTypes from '../actions/actionTypes';
import { updateObject } from '../util/util';

export const initialState = {
  token: null,
  userId: null,
  error: null,
  loading: false,
  registering: false,
  registrationComplete: false,
  authRedirectPath: '/',
};
const authStart = state => updateObject(state, { error: null, loading: true });

const authSuccess = (state, action) => {
  return updateObject(state, {
    token: action.idToken,
    userId: action.userId,
    error: null,
    loading: false,
  });
};

const authFail = (state, action) => {
  return updateObject(state, {
    error: action.error,
    loading: false,
  });
};

const authLogout = state => {
  return updateObject(state, { token: null, userId: null });
};

const beginFormErrorChecking = state => {
  return updateObject(state, {
    error: null,
  });
};

const setFormErrorStart = state => {
  return updateObject(state, {
    error: null,
  });
};

const setFormErrorEnd = (state, action) => {
  return updateObject(state, {
    error: action.error,
  });
};

const registrationStart = state => {
  return updateObject(state, {
    error: null,
    loading: true,
    registering: true,
    registrationComplete: false,
  });
};

const registrationSuccess = state => {
  return updateObject(state, {
    error: null,
    loading: false,
    registering: false,
    registrationComplete: true,
  });
};

const registrationFail = (state, action) => {
  return updateObject(state, {
    error: action.error,
    loading: false,
    registering: false,
    registrationComplete: false,
  });
};

const resetRegistrationCompleted = state => {
  return updateObject(state, {
    registrationComplete: false,
  });
};

const setAuthRedirectPath = (state, action) => {
  return updateObject(state, { authRedirectPath: action.path });
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.LOGIN_START:
      return authStart(state);
    case actionTypes.LOGIN_SUCCESS:
      return authSuccess(state, action);
    case actionTypes.LOGIN_FAIL:
      return authFail(state, action);
    case actionTypes.REGISTRATION_START:
      return registrationStart(state);
    case actionTypes.REGISTRATION_SUCCESS:
      return registrationSuccess(state);
    case actionTypes.REGISTRATION_FAIL:
      return registrationFail(state, action);
    case actionTypes.RESET_REGISTRATION_COMPLETED:
      return resetRegistrationCompleted(state);
    case actionTypes.AUTH_LOGOUT:
      return authLogout(state);
    case actionTypes.BEGIN_FORM_ERROR_CHECKING:
      return beginFormErrorChecking(state);
    case actionTypes.SET_FORM_ERROR_START:
      return setFormErrorStart(state);
    case actionTypes.SET_FORM_ERROR_END:
      return setFormErrorEnd(state, action);
    case actionTypes.SET_AUTH_REDIRECT_PATH:
      return setAuthRedirectPath(state, action);
    default:
      return state;
  }
};

export default reducer;
