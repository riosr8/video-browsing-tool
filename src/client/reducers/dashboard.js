import * as actionTypes from '../actions/actionTypes';
import { updateObject } from '../util/util';

export const initialState = {
  userVideos: [],
  algorithms: [],
  activePage: 1,
  totalPages: 1,
  sortKey: 'time',
  sortOrder: 'asc',
  results: null,
  error: null,
  success: null,
  uploading: false,
  fetching: false,
  fetchingResults: false,
  requestProcessing: false,
  deleting: false,
};

const fetchResultsStart = state => updateObject(state, { error: null, fetchingResults: true });

const fetchResultsSuccess = (state, action) => {
  return updateObject(state, {
    fetchingResults: false,
    results: action.results,
  });
};

const fetchResultsFail = (state, action) => {
  return updateObject(state, {
    error: action.error,
    fetchingResults: false,
  });
};

const fetchVideosStart = state => updateObject(state, { error: null, fetching: true });

const fetchVideosSuccess = (state, action) => {
  return updateObject(state, {
    userVideos: action.userVideos,
    totalPages: action.totalPages,
    fetching: false,
  });
};

const fetchVideosFail = (state, action) => {
  return updateObject(state, {
    error: action.error,
    fetching: false,
  });
};

const uploadVideoStart = state => updateObject(state, { error: null, uploading: true });

const uploadVideoSuccess = (state, action) => {
  return updateObject(state, {
    uploading: false,
    success: action.success,
  });
};

const uploadVideoFail = (state, action) => {
  return updateObject(state, {
    error: action.error,
    uploading: false,
  });
};

const requestProcessingStart = state => {
  return updateObject(state, { error: null, requestProcessing: true });
};

const requestProcessingSuccess = state => {
  return updateObject(state, {
    requestProcessing: false,
  });
};

const requestProcessingFail = (state, action) => {
  return updateObject(state, {
    error: action.error,
    requestProcessing: false,
  });
};

const deleteVideoStart = state => {
  return updateObject(state, {
    deleting: true,
  });
};

const deleteVideoSuccess = (state, action) => {
  return updateObject(state, {
    deleting: false,
    success: action.success,
  });
};

const deleteVideoFail = (state, action) => {
  return updateObject(state, {
    deleting: false,
    error: action.error,
  });
};

const fetchAlgorithmsStart = state => {
  return state;
};

const fetchAlgorithmsSuccess = (state, action) => {
  return updateObject(state, {
    algorithms: action.algorithms,
  });
};

const fetchAlgorithmsFail = (state, action) => {
  return updateObject(state, {
    error: action.error,
  });
};

const clearMessage = state => {
  return updateObject(state, {
    success: null,
    error: null,
  });
};

const setActivePage = (state, action) => {
  return updateObject(state, {
    activePage: action.activePage,
  });
};

const setSortKey = (state, action) => {
  return updateObject(state, {
    sortKey: action.sortKey,
  });
};

const setSortOrder = (state, action) => {
  return updateObject(state, {
    sortOrder: action.sortOrder,
  });
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.FETCH_VIDEOS_START:
      return fetchVideosStart(state);
    case actionTypes.FETCH_VIDEOS_SUCCESS:
      return fetchVideosSuccess(state, action);
    case actionTypes.FETCH_VIDEOS_FAIL:
      return fetchVideosFail(state, action);
    case actionTypes.UPLOAD_VIDEO_START:
      return uploadVideoStart(state);
    case actionTypes.UPLOAD_VIDEO_SUCCESS:
      return uploadVideoSuccess(state, action);
    case actionTypes.UPLOAD_VIDEO_FAIL:
      return uploadVideoFail(state, action);
    case actionTypes.REQUEST_PROCESSING_START:
      return requestProcessingStart(state);
    case actionTypes.REQUEST_PROCESSING_SUCCESS:
      return requestProcessingSuccess(state);
    case actionTypes.REQUEST_PROCESSING_FAIL:
      return requestProcessingFail(state, action);
    case actionTypes.DELETE_VIDEO_START:
      return deleteVideoStart(state);
    case actionTypes.DELETE_VIDEO_SUCCESS:
      return deleteVideoSuccess(state, action);
    case actionTypes.DELETE_VIDEO_FAIL:
      return deleteVideoFail(state, action);
    case actionTypes.FETCH_ALGORITHMS_START:
      return fetchAlgorithmsStart(state);
    case actionTypes.FETCH_ALGORITHMS_SUCCESS:
      return fetchAlgorithmsSuccess(state, action);
    case actionTypes.FETCH_ALGORITHMS_FAIL:
      return fetchAlgorithmsFail(state, action);
    case actionTypes.FETCH_RESULTS_START:
      return fetchResultsStart(state);
    case actionTypes.FETCH_RESULTS_SUCCESS:
      return fetchResultsSuccess(state, action);
    case actionTypes.FETCH_RESULTS_FAIL:
      return fetchResultsFail(state, action);
    case actionTypes.CLEAR_MESSAGE:
      return clearMessage(state);
    case actionTypes.SET_ACTIVE_PAGE:
      return setActivePage(state, action);
    case actionTypes.SET_SORT_KEY:
      return setSortKey(state, action);
    case actionTypes.SET_SORT_ORDER:
      return setSortOrder(state, action);
    case actionTypes.AUTH_LOGOUT:
      return initialState;
    default:
      return state;
  }
};

export default reducer;
