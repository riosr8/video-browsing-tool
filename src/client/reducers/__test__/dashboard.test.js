import reducer, { initialState } from '../dashboard';
import * as actions from '../../actions/dashboard';
import { logout } from '../../actions/auth';

const userVideos = [
  {
    _id: '5c8f0e406745e00852125e82',
    userId: 'test@gmail.com',
    videoName: '437b4a03-d2d7-4862-87e4-55ecf555cb47big_buck_bunny.mp4',
    videoUuid: '437b4a03-d2d7-4862-87e4-55ecf555cb47big_buck_bunny.mp4',
    videoLength: '60.095',
    videoThumbnail: '437b4a03-d2d7-4862-87e4-55ecf555cb47big_buck_bunny.png',
    __v: 0,
  },
];

const results = [
  {
    videoUuid: '437b4a03-d2d7-4862-87e4-55ecf555cb47big_buck_bunny.mp4',
  },
];

const tests = [
  { name: 'should handle FETCH_RESULTS_START', action: actions.fetchResultsStart },
  {
    name: 'should handle FETCH_RESULTS_SUCCESS',
    action: () => {
      return actions.fetchResultsSuccess(results);
    },
  },
  {
    name: 'should handle FETCH_RESULTS_FAIL',
    action: () => {
      return actions.fetchResultsFail('fail');
    },
  },
  { name: 'should handle FETCH_VIDEOS_START', action: actions.fetchVideosStart },
  {
    name: 'should handle FETCH_VIDEOS_SUCCESS',
    action: () => {
      return actions.fetchVideosSuccess(userVideos);
    },
  },
  {
    name: 'should handle FETCH_VIDEOS_FAIL',
    action: () => {
      return actions.fetchVideosFail('fail');
    },
  },
  { name: 'should handle UPLOAD_VIDEO_START', action: actions.uploadVideoStart },
  {
    name: 'should handle UPLOAD_VIDEO_SUCCESS',
    action: () => {
      return actions.uploadVideoSuccess('success');
    },
  },
  {
    name: 'should handle UPLOAD_VIDEO_FAIL',
    action: () => {
      return actions.uploadVideoFail('fail');
    },
  },
  { name: 'should handle REQUEST_PROCESSING_START', action: actions.requestProcessingStart },
  {
    name: 'should handle REQUEST_PROCESSING_SUCCESS',
    action: () => {
      return actions.requestProcessingSuccess('success');
    },
  },
  {
    name: 'should handle REQUEST_PROCESSING_FAIL',
    action: () => {
      return actions.requestProcessingFail('fail');
    },
  },
  { name: 'should handle DELETE_VIDEO_START', action: actions.deleteVideoStart },
  {
    name: 'should handle DELETE_VIDEO_SUCCESS',
    action: () => {
      return actions.deleteVideoSuccess('success');
    },
  },
  {
    name: 'should handle DELETE_VIDEO_FAIL',
    action: () => {
      return actions.deleteVideoFail('fail');
    },
  },
  { name: 'should handle AUTH_LOGOUT', action: logout },
  { name: 'should handle FETCH_ALGORITHMS_START', action: actions.fetchAlgorithmsStart },
  {
    name: 'should handle FETCH_ALGORITHMS_SUCCESS',
    action: () => {
      return actions.fetchAlgorithmsSuccess('success');
    },
  },
  {
    name: 'should handle FETCH_ALGORITHMS_FAIL',
    action: () => {
      return actions.fetchAlgorithmsFail('fail');
    },
  },
  {
    name: 'should handle CLEAR_MESSAGE',
    action: () => {
      return actions.clearMessage();
    },
  },
  {
    name: 'should handle SET_ACTIVE_PAGE',
    action: () => {
      return actions.setActivePage();
    },
  },
  {
    name: 'should handle SET_SORT_KEY',
    action: () => {
      return actions.setSortKey();
    },
  },
  {
    name: 'should handle SET_SORT_ORDER',
    action: () => {
      return actions.setSortOrder();
    },
  },
];

describe('Dashboard reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  tests.forEach(test => {
    it(test.name, () => {
      expect(reducer(initialState, test.action())).toMatchSnapshot();
    });
  });
});
