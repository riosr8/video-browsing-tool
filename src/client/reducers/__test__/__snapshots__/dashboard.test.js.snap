// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard reducer should handle AUTH_LOGOUT 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle CLEAR_MESSAGE 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle DELETE_VIDEO_FAIL 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": "fail",
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle DELETE_VIDEO_START 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": true,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle DELETE_VIDEO_SUCCESS 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": "success",
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle FETCH_ALGORITHMS_FAIL 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": undefined,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle FETCH_ALGORITHMS_START 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle FETCH_ALGORITHMS_SUCCESS 1`] = `
Object {
  "activePage": 1,
  "algorithms": "success",
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle FETCH_RESULTS_FAIL 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": "fail",
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle FETCH_RESULTS_START 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": true,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle FETCH_RESULTS_SUCCESS 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": Array [
    Object {
      "videoUuid": "437b4a03-d2d7-4862-87e4-55ecf555cb47big_buck_bunny.mp4",
    },
  ],
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle FETCH_VIDEOS_FAIL 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": "fail",
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle FETCH_VIDEOS_START 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": true,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle FETCH_VIDEOS_SUCCESS 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": undefined,
  "uploading": false,
  "userVideos": Array [
    Object {
      "__v": 0,
      "_id": "5c8f0e406745e00852125e82",
      "userId": "test@gmail.com",
      "videoLength": "60.095",
      "videoName": "437b4a03-d2d7-4862-87e4-55ecf555cb47big_buck_bunny.mp4",
      "videoThumbnail": "437b4a03-d2d7-4862-87e4-55ecf555cb47big_buck_bunny.png",
      "videoUuid": "437b4a03-d2d7-4862-87e4-55ecf555cb47big_buck_bunny.mp4",
    },
  ],
}
`;

exports[`Dashboard reducer should handle REQUEST_PROCESSING_FAIL 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": "fail",
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle REQUEST_PROCESSING_START 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": true,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle REQUEST_PROCESSING_SUCCESS 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle SET_ACTIVE_PAGE 1`] = `
Object {
  "activePage": undefined,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle SET_SORT_KEY 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": undefined,
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle SET_SORT_ORDER 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": undefined,
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle UPLOAD_VIDEO_FAIL 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": "fail",
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle UPLOAD_VIDEO_START 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": true,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should handle UPLOAD_VIDEO_SUCCESS 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": "success",
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;

exports[`Dashboard reducer should return the initial state 1`] = `
Object {
  "activePage": 1,
  "algorithms": Array [],
  "deleting": false,
  "error": null,
  "fetching": false,
  "fetchingResults": false,
  "requestProcessing": false,
  "results": null,
  "sortKey": "time",
  "sortOrder": "asc",
  "success": null,
  "totalPages": 1,
  "uploading": false,
  "userVideos": Array [],
}
`;
